libnmstate/nm/connection.py

```diff
@@ -321,24 +334,40 @@ def _waitfor_active_connection_callback(
             )
 
     @staticmethod
-    def _add_connection_callback(src_object, result, user_data):
+    def _add_connection2_callback(src_object, result, user_data):
         mainloop = user_data
         try:
-            con = src_object.add_connection_finish(result)
+            profile = src_object.add_connection2_finish(result)[0]
         except Exception as e:
             if mainloop.is_action_canceled(e):
                 logging.debug("Connection adding canceled: error=%s", e)
             else:
                 mainloop.quit("Connection adding failed: error={}".format(e))
             return
 
-        if con is None:
+        if profile is None:
             mainloop.quit("Connection adding failed: error=unknown")
         else:
-            devname = con.get_interface_name()
+            devname = profile.get_interface_name()
             logging.debug("Connection adding succeeded: dev=%s", devname)
             mainloop.execute_next_action()
 
+    @staticmethod
+    def _update2_callback(src_object, result, user_data):
+        mainloop, con_profile = user_data
+        try:
+            src_object.update2_finish(result)
```

Comment:
What about checking if successful or not? `None` seems to represent a (silent) failure.
The documentation is vague, but it does talk about a return value.